ai-agents/fhir-questionnaire-generation/data_ingestion_pipeline/pdf_to_md_service/ftp_utils.py

```
@@ -0,0 +1,116 @@
+import logging
+import os
+import tempfile
+from settings import Configs
+
+async def read_pdf_file_ftp(file_name: str, configs: Configs, logger: logging.Logger):
+    """Read PDF file from FTP server and return local temporary file path."""
+    ftp = None
+    temp_file_path = None
+    try:
+        # Connect to FTP server
+        ftp = ftplib.FTP()
```

Ruff (0.15.15)

[error] FTP-related functions are being called. FTP is considered insecure. Use SSH/SFTP/SCP or some other encrypted protocol.

(S321)